src/vos/tests/vts_io.c

```diff
@@ -2267,16 +2269,33 @@ io_query_key(void **state)
 	assert_int_equal(rc, 0);
 	assert_int_equal(*(uint64_t *)dkey_read.iov_buf, KEY_INC * 2);
 
+	vts_dtx_begin(&oid, arg->ctx.tc_co_hdl, epoch++, 0, &dth);
+
 	rc = vos_obj_query_key(arg->ctx.tc_co_hdl, oid, DAOS_GET_DKEY |
 			       DAOS_GET_MAX, epoch++, &dkey_read, NULL, NULL,
-			       NULL);
+			       dth);
 	assert_int_equal(rc, 0);
 	assert_int_equal(*(uint64_t *)dkey_read.iov_buf, MAX_INT_KEY - KEY_INC);
 
+	vts_dtx_end(dth);
+
+	vts_dtx_begin(&oid, arg->ctx.tc_co_hdl, epoch - 3, 0, &dth);
+	/* Now punch the object at earlier epoch */
```

Comment:
Please move the comment to above line, otherwise it will be some confused because we use "epoch++" as the input parameter for vos_obj_punch().

Honestly, DTX code brings some confused usages for the epoch that may cause the "epoch" parameter for related VOS APIs to be useless. I am think whether we (you or me) should adjust the code to directly use dth->dth_epoch as the "epoch" parameter when call related VOS API?

Reply:
I use the dth_epoch if it's there.    The vos one should probably be documented as "don't care" when there is a dtx